### PR TITLE
UT coverage checking CRC.

### PR DESCRIPTION
**Pull request by BSolonenko**
This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Summary
This PR provides unit test for checking CRC.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by BSolonenko** [comment]
@AKalinich-Luxoft, @AByzhynar, @LuxoftAKutsan, @LitvinenkoIra, @Ypostolov, @ValeriiMalkov, @SNiukalov and @EKuliiev Please review.

File: src/components/application_manager/test/commands/mobile/put_file_test.cc [hunk 1]
```diff
@@ -344,6 +345,53 @@ TEST_F(PutFileRequestTest, Run_InvalidPutFile_UNSUCCESS) {
   command->Run();
 }
 
+TEST_F(PutFileRequestTest, Run_SrcSumEqual_SendSuccessResponse) {
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/put_file_test.cc]: @BSolonenko 
- CRC
- It is not obvious why result is successful. How do we know that binary data is not empty?

File: src/components/application_manager/test/commands/mobile/put_file_test.cc [hunk 2]
```diff
@@ -344,6 +345,53 @@ TEST_F(PutFileRequestTest, Run_InvalidPutFile_UNSUCCESS) {
...
+  (*msg_)[am::strings::msg_params][am::strings::crc32_check_sum] = 2768625435u;
+
+  ExpectReceiveMessageFromSDK();
+  ON_CALL(app_mngr_, get_settings())
+      .WillByDefault(ReturnRef(app_mngr_settings_));
+  ON_CALL(app_mngr_settings_, app_storage_folder())
+      .WillByDefault(ReturnRef(kStorageFolder));
+  ON_CALL(*mock_app_, folder_name()).WillByDefault(Return(kFolder));
+  const uint32_t avaliable_space = binary_data_.size() + 1;
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/put_file_test.cc]: @BSolonenko typo + type `size_t`

File: src/components/application_manager/test/commands/mobile/put_file_test.cc [hunk 3]
```diff
@@ -344,6 +345,53 @@ TEST_F(PutFileRequestTest, Run_InvalidPutFile_UNSUCCESS) {
...
+
+  const std::string file_path = kStorageFolder + "/" + kFolder;
+  EXPECT_CALL(app_mngr_, SaveBinary(binary_data_, file_path, kFileName, 0u))
+      .WillOnce(Return(mobile_apis::Result::SUCCESS));
+  EXPECT_CALL(*mock_app_, increment_put_file_in_none_count());
+  ExpectManageMobileCommandWithResultCode(mobile_apis::Result::SUCCESS);
+  PutFileRequestPtr command(CreateCommand<PutFileRequest>(msg_));
+  ASSERT_TRUE(command->Init());
+  command->Run();
+  EXPECT_TRUE(file_system::RemoveDirectory(kStorageFolder, true));
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/put_file_test.cc]: @BSolonenko Do you create something on file system? Why do you make deletion?

File: src/components/application_manager/test/commands/mobile/put_file_test.cc (hunk 1 above)
Comment (BSolonenko) [reply src/components/application_manager/test/commands/mobile/put_file_test.cc]: @AByzhynar Fixed in https://github.com/smartdevicelink/sdl_core/pull/2139/commits/17cae31dfeb5a8e55281633ebe3efc6dd051c614

File: src/components/application_manager/test/commands/mobile/put_file_test.cc (hunk 2 above)
Comment (BSolonenko) [reply src/components/application_manager/test/commands/mobile/put_file_test.cc]: @AByzhynar Fixed in https://github.com/smartdevicelink/sdl_core/pull/2139/commits/17cae31dfeb5a8e55281633ebe3efc6dd051c614

File: src/components/application_manager/test/commands/mobile/put_file_test.cc (hunk 3 above)
Comment (BSolonenko) [reply src/components/application_manager/test/commands/mobile/put_file_test.cc]: @AByzhynar Fixed in https://github.com/smartdevicelink/sdl_core/pull/2139/commits/17cae31dfeb5a8e55281633ebe3efc6dd051c614

File: src/components/application_manager/test/commands/mobile/put_file_test.cc [hunk 4]
```diff
@@ -345,18 +345,22 @@ TEST_F(PutFileRequestTest, Run_InvalidPutFile_UNSUCCESS) {
   command->Run();
 }
 
-TEST_F(PutFileRequestTest, Run_SrcSumEqual_SendSuccessResponse) {
-  (*msg_)[am::strings::msg_params][am::strings::crc32_check_sum] = 2768625435u;
+TEST_F(PutFileRequestTest, Run_CrcSumEqual_SendSuccessResponse) {
+  binary_data_ = {1u};
+  (*msg_)[am::strings::params][am::strings::binary_data] = binary_data_;
+  uint32_t correct_crc_sum = 2768625435u;
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/put_file_test.cc]: @BSolonenko const and add comment

File: src/components/application_manager/test/commands/mobile/put_file_test.cc [hunk 5]
```diff
@@ -367,28 +371,34 @@ TEST_F(PutFileRequestTest, Run_SrcSumEqual_SendSuccessResponse) {
   PutFileRequestPtr command(CreateCommand<PutFileRequest>(msg_));
   ASSERT_TRUE(command->Init());
   command->Run();
+  // Folder is creating in Run method.
   EXPECT_TRUE(file_system::RemoveDirectory(kStorageFolder, true));
 }
 
-TEST_F(PutFileRequestTest, Run_SrcSumEqual_SendCorruptedDataResponse) {
-  (*msg_)[am::strings::msg_params][am::strings::crc32_check_sum] = 0u;
+TEST_F(PutFileRequestTest, Run_CrcSumEqual_SendCorruptedDataResponse) {
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/put_file_test.cc]: @BSolonenko BAd name for this test case

File: src/components/application_manager/test/commands/mobile/put_file_test.cc [hunk 6]
```diff
@@ -367,28 +371,34 @@ TEST_F(PutFileRequestTest, Run_SrcSumEqual_SendSuccessResponse) {
...
+  binary_data_ = {1u};
+  (*msg_)[am::strings::params][am::strings::binary_data] = binary_data_;
+  uint32_t incorrect_crc_sum = 0u;
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/put_file_test.cc]: @BSolonenko CONST

File: src/components/application_manager/test/commands/mobile/put_file_test.cc [hunk 7]
```diff
@@ -367,28 +371,34 @@ TEST_F(PutFileRequestTest, Run_SrcSumEqual_SendSuccessResponse) {
...
-      .WillByDefault(Return(avaliable_space));
+      .WillByDefault(Return(available_space));
   ON_CALL(*mock_app_, AddFile(_)).WillByDefault(Return(true));
 
   ExpectManageMobileCommandWithResultCode(mobile_apis::Result::CORRUPTED_DATA);
   EXPECT_CALL(app_mngr_, SaveBinary(_, _, _, _)).Times(0);
   PutFileRequestPtr command(CreateCommand<PutFileRequest>(msg_));
   ASSERT_TRUE(command->Init());
   command->Run();
+  // Folder is creating in Run method.
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/put_file_test.cc]: @BSolonenko `is created`

File: src/components/application_manager/test/commands/mobile/put_file_test.cc (hunk 4 above)
Comment (BSolonenko) [reply src/components/application_manager/test/commands/mobile/put_file_test.cc]: @AByzhynar Fixed in https://github.com/smartdevicelink/sdl_core/pull/2139/commits/b05aee0fa7da035c66c5c76041339efd57ee511c

File: src/components/application_manager/test/commands/mobile/put_file_test.cc (hunk 5 above)
Comment (BSolonenko) [reply src/components/application_manager/test/commands/mobile/put_file_test.cc]: @AByzhynar Fixed in b05aee0

File: src/components/application_manager/test/commands/mobile/put_file_test.cc (hunk 6 above)
Comment (BSolonenko) [reply src/components/application_manager/test/commands/mobile/put_file_test.cc]: @AByzhynar Fixed in b05aee0

File: src/components/application_manager/test/commands/mobile/put_file_test.cc (hunk 7 above)
Comment (BSolonenko) [reply src/components/application_manager/test/commands/mobile/put_file_test.cc]: @AByzhynar Fixed in b05aee0

**Comment by BSolonenko** [comment]
@AKalinich-Luxoft Please review.

**Pull request merged**